text/0046-ttfd-automatic-transaction-span.md

```diff
@@ -0,0 +1,90 @@
+- Start Date: 2022-12-16
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/46
+- RFC Status: draft
+  
+# Summary. 
+  
+We want to provide a new span to the automatic UI transactions.  
+The TTFD (time-to-full-display) span is a way for the user to notify the SDK that the UI has been fully loaded.  
+That is, after all data is retrieved, either by database calls or rest APIs, and set into the UI.  
+We would create a new span to the UI automatic transactions to measure it, for all screens of the application.  
+This rfc is mostly about how we should design the new API.  
+  
+# Motivation
+
+This is useful especially for mobile developers, but could also be useful for web.  
+There is no reliable way to automatically detect when UI is fully drawn, as the "fully drawn" concept depends on the developer.  
+  
+# Background. 
+  
+We have to add a new API to the SDK to allow the user to notify that the UI was fully drawn.  
+Also, we need the user to specify the span to finish through a parameter, as it cannot be done automatically.  
+E.g. Activity A starts -> Activity B starts -> Activity A finishes loading data and the API is called.  
+At this point without the activity it was called on, we wouldn't know which span to finish, because the activity B would be at the top of the stack.  
+  
+# Options Considered
+  
+These options were considered for Android, but the same apply to other SDKs, too.  
+1) Add a `SentryAndroid.reportFullyDrawn(Activity)` static method. We would start the span automatically when an Activity is being created and we would finish it when the API is called.  
```

Comment:
Added in [API level 19](https://developer.android.com/guide/topics/manifest/uses-sdk-element#ApiLevels)
So this feature wont work < 19

Reply:
Nope, we would just add a new API to our sdk, without calling system functions. I think it's better to let the user have control over it